doc/developer/animation.md

```diff
@@ -1,50 +1,79 @@
+transformations through the skinnning weights.
 
-## Character Animation
+## Character Animation in Radium
 
-The Radium Engine also provides the basic classes for character animation.
-Starting from a Ra::Core::Animation::Handle, which represents a deformation metaphor, such as Ra::Core::Animation::Cage or Ra::Core::Animation::Skeleton,
-one can associate influence weights to each vertex of a mesh within a `WeightMatrix` (rows representing vertices, columns handles).
+The Radium Engine provides the basic classes for character animation.
+
+Starting from a `HandleSystem`, which represents a deformation metaphor, such as `Cage` or `Skeleton`,
+one must associate influence weights to each vertex of a mesh within a `WeightMatrix`
+(rows representing vertices, columns handles).
 These influence weights, or skinning weights, are then used by a skinning algorithm to deform
-the associated mesh vertices w.r.t. the Ra::Core::Animation::Handle deformation defined by the animation pose.
+the associated mesh vertices w.r.t. the handles deformation defined by the animation pose.
+The mesh is also linked to each handle through a `bindMatrix` that expresses the
+transformation from the mesh's local frame to the handle local frame.
+
+Animations are stored in local space, i.e. each handle transformation is expressed
+in the handle's frame (transformation from bone's parent for skeleton bones,
+position in cage's frame for cage points).
+
+Hence, in order to deform the character's mesh, one must first use the handle's `bindMatrix`
+to express the mesh vertices position into the handles' local space before combining the
+handles local transformations to deform the mesh vertices.
+
+An `Animation` is a set of key poses, each associated to a key time.
+When looking for the pose at a specific animation time, the 2 closest surrounding
+key poses are linearly interpolated to build the querried pose.
+
+Though the `Animation` class only provides keyframing for poses, which are sets
+of transformation matrices, the `KeyFrame` class, unfinished as of now, provides
+a more generic interface, usable for any type of animated data.
+Thus the `Animation` class is tagged as deprecated and will be removed once the `KeyFrame` class is ready.
 
-## The Character Animation Plugins
+## The Skeleton-Based Character Animation Plugin
 
-The **Radium Official Plugins** (https://gitlab.com/Storm-IRIT/radium-official-plugins) provide one plugin specific to skeleton-based character animation
-and another one specific to skeleton-based character skinning.
-This first one manages user interactions with the animation skeleton, enabling posing the character
- and playing the animations.
-The second one is responsible for deforming the object's vertices according to the desired skinning method.
+The **Radium Official Plugins** (https://gitlab.com/Storm-IRIT/radium-official-plugins) provide one plugin specific to skeleton-based character animation and skinning.
```

Comment:
Here you can give the link to the plugin directly.

Reply:
what would you expect?